metricflow/test/integration/test_cases/itest_granularity.yaml

```diff
@@ -25,7 +25,7 @@ integration_test:
       , {{ render_date_trunc("ds", TimeGranularity.WEEK) }} AS metric_time__week
     FROM {{ source_schema }}.fct_bookings_extended
     GROUP BY
-      metric_time__week
+      {{ render_date_trunc("ds", TimeGranularity.WEEK) }}
```

Comment:
I believe this will break BigQuery, because BigQuery does not support function expressions in SELECT and GROUP BY - if you use a function in the SELECT you must group by the alias (or at least that's how it used to be).

If you can update this PR with the other fixes, we can run the test suite against all engines and see if they pass. If they do, there is no need to do anything further here! If, as I suspect, BigQuery doesn't like this change, I think our options are:

1. Use position (`GROUP BY 1, 2, 3` or whatever), as I believe that's supported in all of our current engines
2. Restructure these queries to use subqueries on the select-level transformations (date_trunc and what have you) and then a simple outer query so we can group by the aliases in all cases

I prefer option 2, I always found positional group by a bit unsatisfying, and more importantly it should work with any engine we support in the future (which may not be true of the numeric position label). 

That said, I realize this is a pain in the neck, so please feel free to update to whatever is simplest, and we can update things later as needed.

Reply:
Agree #2 probably better but is going to require quite bit of work. Also, i checked in bigquery and it looks it should not have problem having functions in group by.

![image](https://github.com/dbt-labs/metricflow/assets/138710149/501d3bab-1042-4eca-8681-40be2295bb35)